text/0000-clarity-api-description.md

```diff
@@ -0,0 +1,77 @@
+- Start Date: 2015-06-18
+- RFC PR: (leave this empty)
+- Refract Issue: (leave this empty)
+
+# Summary
+
+Provide clarity and additional elements/properites in API Description namespace.
+
+# Motivation
+
+The purpose of this document is to make it easier for newcomers to understand the format and build tools around it.
+
+# Detailed design
+
+## Combine Namespaces
+
+We should combine the API Description namespace with the resource namespace.
+Right now they are separate and it would be clearer if they were not.
+
+## Elements in Attributes
+
+Refract allows for including elements within the attributes section of an element.
+Even though it’s possible, it may be helpful to avoid this in such an important namespace.
+Reason being, this will be one of the first places people interact with Refract, and to use complex features may be too much.
+
+## HREF Variables
+
+We should call them “hrefVariables” everywhere instead of parameters in places (specifically transitions).
+I think this will be more consistent and easier to understand for newcomers.
+This will make more sense in how we handle inheritance as well, because parameters inheriting from hrefVariables does not seem optimal.
+
+## Defining Classes
+
+Swagger provides a way to define tags that can be applied to resources throughout the document.
+Refract has the “class” attribute that we can use the same way, we only need a way to define a class name and provide the description for it.
+Of course, this won’t carry over to API Blueprint, but that’s OK.
+
+## Clarify Inheritance
+
+We need to clarify how inheritance works by defining the behavior for:
+
+* HREF Variables and HREF values throughout the document
+* Content types in payloads
+* Methods in transitions
+* Resources with schemas (though not in the spec) should be inherited by transitions
+
+## Transitions at Root
+
+We should be able to define transitions on the root of the document and not require they are nested within a resource.
+This type of transition, though, should have a URL and a method.
+
+## Subclass Asset
+
+We should subclass the Asset Element and add one called “messageShema” and another called “messageBody.”
+This will make reading a API Description make more sense than seeing Asset alone with classes.
+
+## Provided Media Types
+
+We should add a section that defines what media types the server will respond with.
+
+## Add Schema to Resource
+
+A resource should be able to have its own schema, whereas now it cannot.
```

Comment:
> Another thing would be if a resource could have multiple assets associated and some of them could be schemas

Sounds great.

Basically, it allows for using the [Resource Model](https://github.com/apiaryio/api-blueprint/blob/master/API%20Blueprint%20Specification.md#def-model-section). Swagger offers simply a way to define the body schema for an entire resource, so this would cover it as well.